src/components/layers-panel/layer-settings-panel.tsx

```diff
@@ -0,0 +1,75 @@
+import { ReactEventHandler } from "react";
+import styled, { DefaultTheme, useTheme } from "styled-components";
+
+import ArrowLeftIcon from "../../../public/icons/arrow-left.svg?svgr";
+import { Sublayer } from "../../types";
+import { CloseButton } from "../close-button/close-button";
+import { BuildingExplorer } from "./building-explorer";
+import { HorizontalLine } from "./horizontal-line";
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  overflow: auto;
+`;
+
+const Header = styled.div<{ theme: DefaultTheme }>`
+  display: flex;
+  margin: 19px 6px 0px;
+  justify-content: space-between;
+  align-items: center;
+  color: ${({ theme }) => theme.colors.fontColor};
+  font-style: normal;
+  font-weight: 700;
+  font-size: 16px;
+  line-height: 19px;
+`;
+
+const Button = styled.div`
+  width: 44px;
+  height: 44px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  cursor: pointer;
+`;
+
+const Icon = styled.div`
+  width: 24px;
+  height: 24px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+export const LayerSettingsPanel = ({
+  sublayers,
+  onUpdateSublayerVisibility,
+  onBackClick,
+  onCloseClick,
+}: {
+  sublayers: Sublayer[];
+  onUpdateSublayerVisibility: (Sublayer) => void;
+  onBackClick: ReactEventHandler;
+  onCloseClick: ReactEventHandler;
+}) => {
+  const theme = useTheme();
+  return (
+    <Container>
+      <Header theme={theme}>
```

Comment:
You don't need to provide theme prop here. It is already accessible from styled component props.

Reply:
Done